web/src/user_profile.js

```diff
@@ -306,6 +303,56 @@ export function get_custom_profile_field_data(user, field, field_types) {
     return profile_field;
 }
 
+function render_user_profile_tab(user) {
+    const $user_profile_name = $(".user_profile_name");
+    $user_profile_name.text(user.full_name);
+
+    const $user_type = $(".top #user-type");
+    const new_user_type = people.get_user_type(user.user_id);
+    $user_type.find(".value").text(new_user_type);
+
+    const field_types = page_params.custom_profile_field_types;
+    const profile_data = page_params.custom_profile_fields
+        .map((f) => get_custom_profile_field_data(user, f, field_types))
+        .filter((f) => f.name !== undefined);
+
+    const $modal_custom_fields = $("#content");
+
+    // Iterate over profileData and render each custom user field
+    for (const field of page_params.custom_profile_fields) {
+        const $existingField = $modal_custom_fields.find(`[data-field-id="${field.id}"]`);
+        $existingField.remove();
+    }
+    const newField = render_user_custom_profile_fields({profile_fields: profile_data});
+    $modal_custom_fields.append(newField);
+}
```

Comment:
Instead of updating each element, we can create a new template file like `user_profile_tab` (we can probably name it better) which will be used in `user_profile_modal.hbs` and here as well.

You can directly pass the args to template and then update in DOM. And then there will probably be no need for separate functions for bots and users.

Reply:
I had difficulty developing a better name so unfortunately, I named the file "user_profile_tab.hbs". If you have any suggestions let me know and I'll be happy to rename it.